lib/exchange/provider/stealth_ex_exchange_provider.dart

```diff
@@ -0,0 +1,259 @@
+import 'dart:convert';
+import 'dart:developer';
+
+import 'package:cake_wallet/.secrets.g.dart' as secrets;
+import 'package:cake_wallet/exchange/provider/exchange_provider.dart';
+import 'package:cake_wallet/exchange/exchange_provider_description.dart';
+import 'package:cake_wallet/exchange/limits.dart';
+import 'package:cake_wallet/exchange/trade.dart';
+import 'package:cake_wallet/exchange/trade_not_created_exception.dart';
+import 'package:cake_wallet/exchange/trade_request.dart';
+import 'package:cake_wallet/exchange/trade_state.dart';
+import 'package:cake_wallet/exchange/utils/currency_pairs_utils.dart';
+import 'package:cw_core/crypto_currency.dart';
+import 'package:http/http.dart' as http;
+
+class StealthExExchangeProvider extends ExchangeProvider {
+  StealthExExchangeProvider() : super(pairList: supportedPairs(_notSupported));
+
+  static const List<CryptoCurrency> _notSupported = [];
+
+  static final apiKey = secrets.stealthExBearerToken;
+  static const _baseUrl = 'https://api.stealthex.io';
+  static const _rangePath = '/v4/rates/range';
+  static const _amountPath = '/v4/rates/estimated-amount';
+  static const _exchangesPath = '/v4/exchanges';
+
+  @override
+  String get title => 'StealthEx';
+
+  @override
+  bool get isAvailable => true;
+
+  @override
+  bool get isEnabled => true;
+
+  @override
+  bool get supportsFixedRate => true;
+
+  @override
+  ExchangeProviderDescription get description => ExchangeProviderDescription.stealthEx;
+
+  @override
+  Future<bool> checkIsAvailable() async => true;
+
+  @override
+  Future<Limits> fetchLimits(
+      {required CryptoCurrency from,
+      required CryptoCurrency to,
+      required bool isFixedRateMode}) async {
+    final headers = {'Authorization': apiKey, 'Content-Type': 'application/json'};
+    final body = {
+      'route': {
+        'from': {'symbol': isFixedRateMode ? to.name : from.name, 'network': 'mainnet'},
+        'to': {'symbol': isFixedRateMode ? from.name : to.name, 'network': 'mainnet'}
```

Comment:
Same question about the network, will we only be dealing with `mainnet`?

Reply:
fixed, thanks for pointing that out.